astropy/units/tests/test_quantity_non_ufuncs.py

```diff
@@ -2521,14 +2520,7 @@ def test_merge_arrays_str(self, flatten):
 )
 tested_functions = get_covered_functions(locals())
 untested_functions = set()
-if NUMPY_LT_1_23:
-    deprecated_functions = {
-        # Deprecated, removed in numpy 1.23
-        np.asscalar,
-        np.alen,
-    }
-else:
-    deprecated_functions = set()
+deprecated_functions = set()
```

Comment:
Do we need `deprecated functions` here and in `astropy/utils/masked/tests/test_function_helpers.py`?

Reply:
Not strictly right now, but I suspect we will again when we really support numpy 2.0 (and 2.1?). So, I'd prefer for this to stay, but do see the case for removing it too (together with the following `untested_functions |= deprecated_functions`) -- really, the whole phrasing of this part was to help keep track of what is not tested and why.

Reply:
On one hand the lines in question would be very simple to replace if the need arises. On the other hand declaring the set of deprecated functions to be empty is very explicit. I would prefer to remove them, but I'm not going to demand it.

Reply:
Up to you, then, @nstarman!